builder/views.py

```diff
@@ -118,13 +118,22 @@ def get(self, request: Request, key: str, **kwargs) -> HttpResponse:
 
         return JsonResponse(self._get(request, course))
 
+    @login_required_method(redirect_url=None)
+    def get(self, request: Request, remote_id: int, **kwargs) -> HttpResponse:
+        course = get_object_or_404(Course, remote_id=remote_id)
+
+        if not course.has_access(request, Permission.READ):
+            return JsonResponse({"success": False, "error": f"No access to instance {course.remote_id}"})
+
+        return JsonResponse(self._get(request, course))
+
```

Comment:
Please combine this with the above method, we don't want to duplicate code. You can do `def get(self, request: Request, key: Optional[str] = None, remote_id: Optional[int] = None, **kwargs):`, and then check which parameter is not None.